README.adoc

```diff
@@ -0,0 +1,190 @@
+
+* A text editor or IDE
+* Java 8 higher
+* Maven
+
+== Create a project
+
+Here is the content of the `pom.xml` file you should be using:
+
+[source,xml]
+.Maven `pom.xml`
+----
+include::pom.xml[]
+----
+
+== Load the OpenAPI document
+
+Vert.x Web API Contract provides you `OpenAPI3RouterFactory`, an object that helps you to build the Vert.x Web Router starting from the OpenAPI specification.
+
+To load the specification into the start method of your Verticle:
+[source,java]
+----
+@Override
+public void start(Future<Void> future) {
+  OpenAPI3RouterFactory.create(this.vertx, "petstore.yaml", ar -> {
+    if (ar.succeeded()) {
+      OpenAPI3RouterFactory routerFactory = ar.result(); <1>
+    } else {
+      // Something went wrong during router factory initialization
+      future.fail(ar.cause()); // <2>
+    }
+  });
+}
+----
+<1> If the loading succeeds, you receive a ready to use instance of `OpenAPI3RouterFactory`, otherwise
+<2> you fail the deploy of the verticle
+
+== Write the handlers
+
+Now you can fit your business logic into the Route handlers using `addHandlerByOperationId()`.
+
+For `listPets`:
+[source,java]
+----
+routerFactory.addHandlerByOperationId("listPets", routingContext ->
+  routingContext
+    .response() // <1>
+    .setStatusCode(200)
+    .putHeader(HttpHeaders.CONTENT_TYPE, "application/json") // <2>
+    .end(new JsonArray(getAllPets()).encode()) // <3>
+);
+----
+<1> Get the response object
+<2> Put `Content-type: application/json` header
+<3> Write the response with all pets
+
+For `createPets`:
+[source,java]
+----
+routerFactory.addHandlerByOperationId("createPets", routingContext -> {
+  RequestParameters params = routingContext.get("parsedParameters"); // <1>
+  JsonObject pet = params.body().getJsonObject(); // <2>
+  addPet(pet);
+  routingContext
+    .response()
+    .setStatusCode(200)
+    .end(); // <3>
+});
+----
+<1> Get the parsed parameters container
+<2> Extract the parsed body
+<3> Write the 200 empty response
+
+For `showPetById`:
+[source,java]
+----
+routerFactory.addHandlerByOperationId("showPetById", routingContext -> {
+  RequestParameters params = routingContext.get("parsedParameters"); // <1>
+  Integer id = params.pathParameter("petId").getInteger(); // <2>
+  Optional<JsonObject> pet = getAllPets().stream().filter(p -> p.getInteger("id").equals(id)).findFirst(); // <3>
+  if (pet.isPresent())
+    routingContext
+      .response()
+      .setStatusCode(200)
+      .putHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+      .end(pet.get().encode()); // <4>
+  else
+    routingContext.fail(404, new Exception("Pet not found")); // <5>
+});
+----
+<1> Get the parsed parameters container
+<2> Extract the parsed path parameter
+<3> Search the pet
+<4> If pet is present, write the pet in the response
+<5> If pet is absent, fail the routing context with 404
+
+== Get the router
+
+Now we can generate the `Router` and add the "Not Found" and "Bad Request" error handlers:
+[source,java]
```

Comment:
You should include code snippets using Asciidoc `include`, possibly with tags so you don't include the whole file but smaller bits.